src/Components/Web.Extensions/src/InputFile/RemoteBrowserFileStream.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal class RemoteBrowserFileStream : BrowserFileStream
+    {
+        private readonly IJSRuntime _jsRuntime;
+        private readonly ElementReference _inputFileElement;
+        private readonly int _maxChunkSize;
+        private readonly PipeReader _pipeReader;
+        private readonly CancellationTokenSource _fillBufferCts;
+
+        private bool _isReadingCompleted;
+        private bool _isDisposed;
+
+        public RemoteBrowserFileStream(IJSRuntime jsRuntime, ElementReference inputFileElement, int maxChunkSize, int maxBufferSize, BrowserFile file)
+            : base(file)
+        {
+            _jsRuntime = jsRuntime;
+            _inputFileElement = inputFileElement;
+            _maxChunkSize = maxChunkSize;
+
+            var pipe = new Pipe(new PipeOptions(pauseWriterThreshold: maxBufferSize, resumeWriterThreshold: maxBufferSize));
```

Comment:
@jkotalik \ @halter73 \ @davidfowl  could I have one of you help review this code?

Reply:
Does this use the default memory pool? Should it use it's own memory pool instead? If the minimumSegmentSize doesn't align with the underlying implementation `page/chunk` size this will result in a lot of arrays being created, I think.

@davidfowl @halter73 thoughts?